cli/internal/terraform/terraform.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package terraform
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/edgelesssys/constellation/internal/cloud/cloudprovider"
+	"github.com/edgelesssys/constellation/internal/file"
+	"github.com/edgelesssys/constellation/internal/state"
+	"github.com/hashicorp/go-version"
+	install "github.com/hashicorp/hc-install"
+	"github.com/hashicorp/hc-install/fs"
+	"github.com/hashicorp/hc-install/product"
+	"github.com/hashicorp/hc-install/releases"
+	"github.com/hashicorp/hc-install/src"
+	"github.com/hashicorp/terraform-exec/tfexec"
+	tfjson "github.com/hashicorp/terraform-json"
+	"github.com/spf13/afero"
+)
+
+const (
+	tfVersion         = ">= 1.2.0"
```

Comment:
Guess we should pin a concrete version.

Reply:
The reason for specifying a version constraint instead of pinning a concrete version is to avoid downloading Terraform if a user already has a newer version installed (which should be backwards compatible).

Newest release is `v1.2.9` not sure if we want to set a higher minimum version, but `v1.2.0` works for creating qemu clusters.